scripts/benchmark_db/lotus_bench.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+require_relative 'benchmark_base'
+
+# Lotus benchmark class
+class LotusBenchmark < BenchmarkBase
```

Comment:
I was not able to run Lotus benchmark:

```
I, [2023-05-22T11:55:06.469437 #55451]  INFO -- : Running bench: lotus
I, [2023-05-22T11:55:06.469465 #55451]  INFO -- : Building artefacts...
I, [2023-05-22T11:55:06.469478 #55451]  INFO -- : Cloning repository
I, [2023-05-22T11:55:06.469487 #55451]  INFO -- : $ git clone https://github.com/filecoin-project/lotus.git lotus
Cloning into 'lotus'...
remote: Enumerating objects: 148017, done.
remote: Total 148017 (delta 0), reused 0 (delta 0), pack-reused 148017
Receiving objects: 100% (148017/148017), 257.16 MiB | 21.76 MiB/s, done.
Resolving deltas: 100% (108200/108200), done.
I, [2023-05-22T11:55:20.750001 #55451]  INFO -- : Clean and build client
I, [2023-05-22T11:55:20.750320 #55451]  INFO -- : $ git checkout master
Already on 'master'
Your branch is up to date with 'origin/master'.
I, [2023-05-22T11:55:21.278269 #55451]  INFO -- : $ make clean
rm -rf  build/.filecoin-install build/.update-modules  lotus lotus-miner lotus-worker lotus-shed lotus-gateway lotus-seed lotus-fountain lotus-bench lotus-stats lotus-pcr lotus-health lotus-wallet lotus-keygen testground tvx lotus-sim
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C extern/filecoin-ffi/ clean
make[1]: *** No rule to make target `clean'.  Stop.
make: [clean] Error 2 (ignored)
I, [2023-05-22T11:55:21.803503 #55451]  INFO -- : $ make calibnet
git submodule update --init --recursive
Submodule 'extern/filecoin-ffi' (https://github.com/filecoin-project/filecoin-ffi.git) registered for path 'extern/filecoin-ffi'
Submodule 'extern/serialization-vectors' (https://github.com/filecoin-project/serialization-vectors.git) registered for path 'extern/serialization-vectors'
Submodule 'extern/test-vectors' (https://github.com/filecoin-project/test-vectors.git) registered for path 'extern/test-vectors'
Cloning into '/Users/guillaume/Code/forest/benchdir/lotus/extern/filecoin-ffi'...
Cloning into '/Users/guillaume/Code/forest/benchdir/lotus/extern/serialization-vectors'...
Cloning into '/Users/guillaume/Code/forest/benchdir/lotus/extern/test-vectors'...
Submodule path 'extern/filecoin-ffi': checked out 'de34caff946d598edb299566d951b44b9b7f7dd4'
Submodule path 'extern/serialization-vectors': checked out '5bfb928910b01ac8b940a693af2884f7f8276211'
Submodule path 'extern/test-vectors': checked out '28b0c45eab4c302864af0aeaaff813625cfafe97'
Submodule 'gen/extern/fil-blst' (https://github.com/filecoin-project/fil-blst.git) registered for path 'extern/test-vectors/gen/extern/fil-blst'
Submodule 'gen/extern/filecoin-ffi' (https://github.com/filecoin-project/filecoin-ffi.git) registered for path 'extern/test-vectors/gen/extern/filecoin-ffi'
Cloning into '/Users/guillaume/Code/forest/benchdir/lotus/extern/test-vectors/gen/extern/fil-blst'...
Cloning into '/Users/guillaume/Code/forest/benchdir/lotus/extern/test-vectors/gen/extern/filecoin-ffi'...
Submodule path 'extern/test-vectors/gen/extern/fil-blst': checked out '5f93488fc0dbfb450f2355269f18fc67010d59bb'
Submodule path 'extern/test-vectors/gen/extern/filecoin-ffi': checked out 'f640612a1a1f7a2dd8b3a49e1531db0aa0f63447'
touch build/.update-modules
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C extern/filecoin-ffi/ .install-filcrypto
go clean -cache -testcache
./install-filcrypto
+ auth_header=()
+ '[' -n '' ']'
++ dirname ./install-filcrypto
+ cd .
+ rust_sources_dir=rust
++ jq -r '.[].check_cpu_for_feature | select(. != null)'
+ cpu_features_required_for_optimized_release='adx
sha_ni
sha2
sse2
avx2
avx
sse4_2
sse4_1'
+ main
++ get_release_flags
++ local __features=
++ [[ ! -f /proc/cpuinfo ]]
++ echo '[get_release_flags] no /proc/cpuinfo file; falling back to Darwin feature detection'
[get_release_flags] no /proc/cpuinfo file; falling back to Darwin feature detection
+++ sysctl -a
+++ grep machdep.cpu
+++ tr '[:upper:]' '[:lower:]'
+++ grep features
++ __features=
+++ uname -s
++ '[' Darwin = Darwin ']'
+++ uname -m
++ '[' arm64 = x86_64 ']'
++ feature_map=("adx:+adx" "sha_ni:+sha" "sha2:+sha2" "sse2:+sse2" "avx2:+avx2" "avx:+avx" "sse4_2:+sse4.2" "sse4_1:+sse4.1")
++ target_features=
++ for x in '${cpu_features_required_for_optimized_release[@]}'
+++ echo ''
+++ grep -c adx
++ current_feature=0
++ '[' 1 = 0 ']'
++ for x in '${cpu_features_required_for_optimized_release[@]}'
+++ echo ''
+++ grep -c sha_ni
++ current_feature=0
++ '[' 1 = 0 ']'
++ for x in '${cpu_features_required_for_optimized_release[@]}'
+++ echo ''
+++ grep -c sha2
++ current_feature=0
++ '[' 1 = 0 ']'
++ for x in '${cpu_features_required_for_optimized_release[@]}'
+++ echo ''
+++ grep -c sse2
++ current_feature=0
++ '[' 1 = 0 ']'
++ for x in '${cpu_features_required_for_optimized_release[@]}'
+++ echo ''
+++ grep -c avx2
++ current_feature=0
++ '[' 1 = 0 ']'
++ for x in '${cpu_features_required_for_optimized_release[@]}'
+++ echo ''
+++ grep -c avx
++ current_feature=0
++ '[' 1 = 0 ']'
++ for x in '${cpu_features_required_for_optimized_release[@]}'
+++ echo ''
+++ grep -c sse4_2
++ current_feature=0
++ '[' 1 = 0 ']'
++ for x in '${cpu_features_required_for_optimized_release[@]}'
+++ echo ''
+++ grep -c sse4_1
++ current_feature=0
++ '[' 1 = 0 ']'
++ echo ''
+ local __release_flags=
+ '[' 1 '!=' 1 ']'
+ echo '[install-filcrypto/main] building libfilcrypto from local sources (dir = rust)'
[install-filcrypto/main] building libfilcrypto from local sources (dir = rust)
+ build_from_source filcrypto rust ''
+ local __library_name=filcrypto
+ local __rust_sources_path=rust
+ local __release_flags=
++ git rev-parse HEAD
+ local __repo_sha1=de34caff946d598edb299566d951b44b9b7f7dd4
+ local __repo_sha1_truncated=de34caff946d598e
+ echo 'building from source @ de34caff946d598e'
building from source @ de34caff946d598e
++ command -v cargo
+ '[' -x /Users/guillaume/.cargo/bin/cargo ']'
++ command -v rustup
+ '[' -x /Users/guillaume/.cargo/bin/rustup ']'
+ pushd rust
+ cargo --version
~/Code/forest/benchdir/lotus/extern/filecoin-ffi/rust ~/Code/forest/benchdir/lotus/extern/filecoin-ffi
cargo 1.62.0-nightly (a44758ac8 2022-05-04)
+ additional_flags=
++ uname -s
+ '[' Darwin = Darwin ']'
++ uname -m
+ '[' arm64 = x86_64 ']'
+ build=build
+ '[' '' == 0 ']'
+ gpu_flags=,cuda
+ '[' '' == 1 ']'
++ uname -s
+ '[' Darwin = Darwin ']'
+ gpu_flags=,opencl
+ use_multicore_sdr=multicore-sdr
+ '[' '' == 0 ']'
+ '[' '' == 1 ']'
+ additional_flags=' --no-default-features --features multicore-sdr,opencl'
+ echo 'Using additional build flags:  --no-default-features --features multicore-sdr,opencl'
+ '[' -n '' ']'
Using additional build flags:  --no-default-features --features multicore-sdr,opencl
++ cat rust-toolchain
+ ./scripts/build-release.sh filcrypto nightly-2022-05-09 build ' --no-default-features --features multicore-sdr,opencl'
+ main filcrypto nightly-2022-05-09 build ' --no-default-features --features multicore-sdr,opencl'
+ [[ -z filcrypto ]]
+ [[ -z nightly-2022-05-09 ]]
+ [[ -z build ]]
++ mktemp
+ __build_output_log_tmp=/var/folders/nt/lktvswgx2qxb65dsy45p_p2c0000gn/T/tmp.PDmXfWczJP
+ trap '{ rm -f $__build_output_log_tmp; }' EXIT
+ rustup target add --toolchain nightly-2022-05-09 wasm32-unknown-unknown
info: component 'rust-std' for target 'wasm32-unknown-unknown' is up to date
+ local '__rust_flags=--print native-static-libs '
+ RUSTFLAGS='--print native-static-libs '
+ cargo +nightly-2022-05-09 build --release --locked --no-default-features --features multicore-sdr,opencl
+ tee /var/folders/nt/lktvswgx2qxb65dsy45p_p2c0000gn/T/tmp.PDmXfWczJP
error: failed to load manifest for workspace member `/Users/guillaume/Code/forest/forest/cli`

Caused by:
  failed to parse manifest at `/Users/guillaume/Code/forest/forest/cli/Cargo.toml`

Caused by:
  feature `workspace-inheritance` is required

  The package requires the Cargo feature called `workspace-inheritance`, but that feature is not stabilized in this version of Cargo (1.62.0-nightly (a44758ac8 2022-05-04)).
  Consider adding `cargo-features = ["workspace-inheritance"]` to the top of Cargo.toml (above the [package] table) to tell Cargo you are opting in to use this unstable feature.
  See https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#workspace-inheritance for more information about the status of this feature.
+ rm -f /var/folders/nt/lktvswgx2qxb65dsy45p_p2c0000gn/T/tmp.PDmXfWczJP
make[1]: *** [.install-filcrypto] Error 101
make: *** [build/.filecoin-install] Error 2
E, [2023-05-22T11:55:42.755297 #55451] ERROR -- : Fiasco during benchmark run. Deleting downloaded files and stopping process...
```

Reply:
I haven't had this issue on my machine or on any of the droplets. Is this on your local machine with `rustup` already installed? If so, try running `rustup udpate` (see this issue). If that fixes it, we might want to add a note to the `README.md` (for users who already have Rust installed) to run `rustup update`.

Reply:
I've found the issue, I was using an in-forest-repo tempdir:
`./scripts/benchmark_db/bench.rb ../../Snapshots/462240_2023_04_11T06_13_00Z.car --daily --chain calibnet --tempdir benchdir`

Using out-of-forest tempdir fixed the issue (cargo was trying loading some of our manifest):
`./scripts/benchmark_db/bench.rb ../../Snapshots/462240_2023_04_11T06_13_00Z.car --daily --chain calibnet --tempdir ../benchdir`

I think we can ignore that at the moment because we will run the script from some iac repo.